### PR TITLE
refactor(boot): increases wait timeout for servlets initialization

### PR DESCRIPTION
**Pull request by PatrickfBraz**
## Context 

Typically in my deployment environment, servlets take anywhere from 20 seconds to 40 seconds to start. Because of this, in addition to the fact that I use more than one GMS replica, at times not all pods start correctly, resulting in the failure to deploy the Helm package to the Kubernetes cluster. Therefore, increasing the servlet initialization timeout can solve this problem for other developers who may experience the same problems.

### Logs example:

```
[main] INFO  o.s.web.servlet.DispatcherServlet:547 - Completed initialization in 3709 ms
INFO:oejshC.ROOT:main: Initializing Spring DispatcherServlet 'authServlet'
[main] WARN  c.l.m.boot.OnBootApplicationListener:50 - OnBootApplicationListener context refreshed! false event: org.springframework.context.event.ContextRefreshedEvent[source=WebApplicationContext for namespace 'authServlet-servlet', started on Wed Oct 04 12:44:57 GMT 2023, parent: Root WebApplicationContext]
...
[main] INFO  o.s.web.servlet.DispatcherServlet:547 - Completed initialization in 16405 ms
INFO:oejshC.ROOT:main: Initializing Spring DispatcherServlet 'openapiServlet'
[main] INFO  o.s.web.servlet.DispatcherServlet:525 - Initializing Servlet 'openapiServlet'
...
[main] INFO  o.s.web.servlet.DispatcherServlet:547 - Completed initialization in 11981 ms
main: Initializing Spring DispatcherServlet 'schemaRegistryServlet'
[main] INFO  o.s.web.servlet.DispatcherServlet:525 - Initializing Servlet 'schemaRegistryServlet'
...
[main] WARN  c.l.m.boot.OnBootApplicationListener:50 - OnBootApplicationListener context refreshed! false event: org.springframework.context.event.ContextRefreshedEvent[source=WebApplicationContext for namespace 'schemaRegistryServlet-servlet', started on Wed Oct 04 12:45:26 GMT 2023, parent: Root WebApplicationContext]
[main] INFO  o.s.web.servlet.DispatcherServlet:547 - Completed initialization in 6409 ms
INFO:oejsh.ContextHandler:main: Started o.e.j.w.WebAppContext@5f058f00{Open source GMS,/,[file:///tmp/jetty-0_0_0_0-8080-war_war-_-any-10472262629014418870/webapp/, jar:file:///tmp/jetty-0_0_0_0-8080-war_war-_-any-10472262629014418870/webapp/WEB-INF/lib/swagger-ui-4.15.5.jar!/META-INF/resources],AVAILABLE}{file:///datahub/datahub-gms/bin/war.war}
INFO:oejs.AbstractConnector:main: Started ServerConnector@7fc229ab{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
INFO:oejs.Server:main: Started @190466ms
[pool-13-thread-1] INFO  c.l.m.boot.OnBootApplicationListener:73 - Connected! Authentication not tested.
[pool-13-thread-1] INFO  c.l.metadata.boot.BootstrapManager:26 - Starting Bootstrap Process...
[pool-13-thread-1] INFO  c.l.metadata.boot.BootstrapManager:33 - Executing bootstrap step 1/15 with name WaitForSystemUpdateStep...
```

NOTE: Sometimes the initialization fails even before the `schemaRegistryServlet` starts.

## Tests

Currently, my production environment already has this change and is working perfectly.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/factories/src/main/java/com/linkedin/metadata/boot/OnBootApplicationListener.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public void onApplicationEvent(@Nonnull ContextRefreshedEvent event) {
   public Runnable isSchemaRegistryAPIServletReady() {
     return () -> {
         final HttpGet request = new HttpGet(provider.getKafka().getSchemaRegistry().getUrl());
-        int timeouts = 30;
+        int timeouts = 60;
```
Comment (RyanHolstien) [inline metadata-service/factories/src/main/java/com/linkedin/metadata/boot/OnBootApplicationListener.java:65]: Can this get pushed into a configuration? 😄

Comment (PatrickfBraz) [reply metadata-service/factories/src/main/java/com/linkedin/metadata/boot/OnBootApplicationListener.java:65]: I think so.

**Comment by PatrickfBraz** [comment]
Now I added the possibility of being configured via environment variable

**Review by RyanHolstien — APPROVED** [review]
LGTM, will merge once CI is green 😄 thanks!

**Pull request merged**